ai/default/aiunit.cpp

```diff
@@ -640,6 +640,8 @@ static void dai_military_bodyguard(struct ai_type *ait,
       int them2goal = real_map_distance(unit_tile(aunit), aunit->goto_tile);
       int unit_mv_rate = unit_move_rate(punit);
       fc_assert_ret_msg(unit_mv_rate, "div by zero");
+      fc_assert_ret_msg(punit, "div by zero");
+      fc_assert_ret_msg(aunit, "div by zero");
```

Comment:
This is not sufficient, even existing units can have a move rate of 0 if `min_speed` is 0 in `units.ruleset`. Consider the following addition to the classic ruleset after setting `min_speed = 0` for Land units:
```ini
; Units get stuck in forests
[effect_elves_catch_units]
type    = "Move_Bonus"
value	= -1
reqs	=
    { "type", "name", "range"
      "Terrain", "Forest", "Local"
    }
```
Now try to play this from the beginning... (and notice the many bugs it exposes)

=> you need to check the move rate as is already done for `punit`